dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Intrinsics.java

```diff
@@ -0,0 +1,784 @@
+            return variants.stream().anyMatch(v -> matchingFunction.test(funcName, v));
+        }
+    }
+
+    @FunctionalInterface
+    private interface Replacer {
+        List<Event> replace(FunctionCall call);
+    }
+
+    private final List<Info> INTRINSICS = ImmutableList.copyOf(List.of(
+            // --------------------------- pthread threading ---------------------------
+            new Info("pthread_create", true, false, true, false, null),
+            new Info("pthread_exit", false, false, false, false, null),
+            new Info("pthread_join",
+                    List.of("pthread_join", "__pthread_join", "\"\\01_pthread_join\""),
+                    false, true, false, false, null),
+            new Info("pthread_barrier_wait", false, false, true, true, this::inlineAsZero),
+            // --------------------------- pthread mutex ---------------------------
+            new Info("pthread_mutex_init", true, false, true, true, this::inlinePthreadMutexInit),
+            new Info("pthread_mutex_lock", true, true, false, true, this::inlinePthreadMutexLock),
+            new Info("pthread_mutex_unlock", true, false, true, true, this::inlinePthreadMutexUnlock),
+            new Info("pthread_mutex_destroy", true, false, true, true, this::inlinePthreadMutexDestroy),
+            // --------------------------- SVCOMP ---------------------------
+            new Info("__VERIFIER_atomic_begin", false, false, true, true, this::inlineAtomicBegin),
+            new Info("__VERIFIER_atomic_end", false, false, true, true, this::inlineAtomicEnd),
+            // --------------------------- __VERIFIER ---------------------------
+            new Info("__VERIFIER_loop_begin", false, false, true, true, this::inlineLoopBegin),
+            new Info("__VERIFIER_loop_bound", false, false, true, true, this::inlineLoopBound),
+            new Info("__VERIFIER_spin_start", false, false, true, true, this::inlineSpinStart),
+            new Info("__VERIFIER_spin_end", false, false, true, true, this::inlineSpinEnd),
+            new Info("__VERIFIER_assume", false, false, true, true, this::inlineAssume),
+            new Info("__VERIFIER_assert", false, false, false, false, this::inlineAssert),
+            new Info("__VERIFIER_nondet",
+                    List.of("__VERIFIER_nondet_bool",
+                            "__VERIFIER_nondet_int", "__VERIFIER_nondet_uint", "__VERIFIER_nondet_unsigned_int",
+                            "__VERIFIER_nondet_short", "__VERIFIER_nondet_ushort", "__VERIFIER_nondet_unsigned_short",
+                            "__VERIFIER_nondet_long", "__VERIFIER_nondet_ulong",
+                            "__VERIFIER_nondet_char", "__VERIFIER_nondet_uchar"),
+                    false, false, true, false, this::inlineNonDet),
+            // --------------------------- LLVM ---------------------------
+            new Info("llvm.*",
```

Comment:
Why do we have `*` here as "matching everything" but not in other ones have several variants? e.g. nondet

Reply:
The meaning is different. `*` means that each listed variant is an incomplete name, because, e.g., all the llvm intrinsics will have a suffix like `.i32.i64`. These suffixes are not present in the `__VERIFIER_nondet` case, meaning that all listed names have to match exactly.